components/src/dynamo/vllm/multimodal_utils/model.py

```
@@ -33,6 +35,30 @@
+def _maybe_skip_encoder_only_kernel_warmup() -> Iterator[None]:
+    if os.getenv("DYN_VLLM_SKIP_ENCODER_ONLY_KERNEL_WARMUP") != "1":
+        yield
+        return
+
+    # vLLM imports kernel_warmup directly into gpu_worker, so patch the symbol
+    # used by GPUWorker.compile_or_warm_up_model. Encoder-only workers do not
+    # execute generation kernels; their vision encoder is warmed by real image
+    # requests instead. Keep this patch scoped to synchronous LLM construction
+    # and restore it even when initialization fails.
+    worker_module = import_module("vllm.v1.worker.gpu_worker")
+    original = getattr(worker_module, "kernel_warmup")
```

Ruff (0.16.2)

[warning] Do not call `getattr` with a constant attribute value. It is not any safer than normal property access.

Replace `getattr` with attribute access

(B009)